### ISSUE TITLE
Add support for optional fields for io-ts and Sarcastic.

### ISSUE THREAD

**Issue by stereobooster** [transform-it/transform-json-types#17]
I realised I haven't implemented it, because in my tests I do not use optional fields.

This is not correct JS:

```js
const Hello = t.type( {
  type: t.number,
  longitude: t.number,
  tags: t.array(t.string),
  friends: t.array(Friends),
  optionalField?: t.string, // <-- here
});
```

**Comment by ritz078** [comment]
fixed in 0.5.2

**Comment by stereobooster** [comment]
I guess this is closest alternative for `io-ts` I guess

```ts
type Hello = {
  optionalField: string | undefined
}
```

```js
const Hello = t.type({
  optionalField: t.union([t.string, t.undefined])
})
```
